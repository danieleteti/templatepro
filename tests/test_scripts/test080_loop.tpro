************* DataSets
{{for cust in customers}}
{{:cust.@@index}} Code: {{:cust.Code}} - Name: {{:cust.Name}}
{{if cust.@@odd }}this line is odd{{endif}}
{{if cust.@@even }}this line is even{{endif}}

{{endfor}}

************* List of objects
{{for obj in objects}}
Prop1: {{:obj.Prop1}} {{:value2}}
{{:obj.@@index}}
{{if obj.@@odd }}this line is odd{{endif}}
{{if obj.@@even }}this line is even{{endif}}

---
{{endfor}}

************* reset DataSet
{{ for cust in customers }}
{{: cust.@@index }}
{{ endfor }}

************* reset List of objects
{{for obj in objects}}
{{:obj.@@index}}
{{endfor}}

*************check for empty dataset (o rows)
{{for ds in emptydataset}}
NEVER SHOWN
{{endfor}}

{{if emptydataset}}
wrong
{{endif}}
{{if !emptydataset}}
ok
{{endif}}
{{if emptydataset}}
wrong
{{else}}
ok
{{endif}}

*************check for undefined variables
{{if blabla}}
wrong
{{endif}}
{{if !blabla}}
ok
{{endif}}
{{if blabla}}
wrong
{{else}}
ok
{{endif}}
