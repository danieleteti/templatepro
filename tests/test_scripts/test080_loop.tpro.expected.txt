************* DataSets
1 Code: 1 - Name: Ford
this line is odd
2 Code: 2 - Name: Ferrari
this line is even
3 Code: 3 - Name: Lotus
this line is odd
4 Code: 4 - Name: FCA
this line is even
5 Code: 5 - Name: Hyundai
this line is odd
6 Code: 6 - Name: De Tomaso
this line is even
7 Code: 7 - Name: Dodge
this line is odd
8 Code: 8 - Name: Tesla
this line is even
9 Code: 9 - Name: Kia
this line is odd
10 Code: 10 - Name: Tata
this line is even
11 Code: 11 - Name: Volkswagen
this line is odd
12 Code: 12 - Name: Audi
this line is even
13 Code: 13 - Name: Skoda
this line is odd

************* List of objects
Prop1: value1.1 DANIELE2
1
this line is odd
---
Prop1: value1.2 DANIELE2
2
this line is even
---
Prop1: value1.3 DANIELE2
3
this line is odd
---

************* reset DataSet
1
2
3
4
5
6
7
8
9
10
11
12
13

************* reset List of objects
1
2
3

*************check for empty dataset (o rows)

ok
ok

*************check for undefined variables
ok
ok
